TFEngine.py

```diff
@@ -1857,86 +1857,127 @@ def search(self, dataset, do_eval=True, output_layer_name="output", output_file=
       max_seq_length=max_seq_length,
       used_data_keys=self.network.used_data_keys)
 
-    output_layer = self.network.layers[output_layer_name]
-    out_beam_size = output_layer.output.beam_size
-    output_layer_beam_scores = None
-    if out_beam_size is None:
-      print("Given output %r is after decision (no beam)." % output_layer, file=log.v1)
-    else:
-      print("Given output %r has beam size %i." % (output_layer, out_beam_size), file=log.v1)
-      output_layer_beam_scores = output_layer.get_search_choices().beam_scores
-    target_key = output_layer.target or self.network.extern_data.default_target
+    if not isinstance(output_layer_names, list):
+      output_layer_names = [output_layer_names]
+    num_targets = len(output_layer_names)
+
+    output_layers, out_beam_sizes, output_layer_beam_scores, target_keys = [], [], [], []
```

Comment:
There should be a comment about the size of the lists.

Reply:
Oh, forgot that one. Do you mean just a comment saying all lists will be of length num_targets? Or do you mean the data type?

Reply:
The data types would probably be clearer (it's not totally obvious in all the cases), but also the length. So maybe make each assign an individual line also.